workflows/runCI.js

```diff
@@ -3,8 +3,2717 @@
+
+  var who = bag.who + '|' + _saveStepState.name;
+  logger.verbose(who, 'Inside');
+
+  bag.consoleAdapter.openGrp('Saving Job Files');
+
+  saveState(bag,
+    function (err, sha) {
+      if (err) {
+        logger.error(who,
+          util.format('Failed to save state for resource: %s',
+            bag.inPayload.name), err
+        );
+
+        bag.consoleAdapter.closeGrp(false);
+
+        bag.jobStatusCode = getStatusCodeByName('FAILED');
+      } else {
+        bag.versionSha = sha;
+        bag.consoleAdapter.closeGrp(true);
+      }
+      return next();
+    }
+  );
+}
+
+function _getOutputVersion(bag, next) {
+  if (bag.isJobCancelled) return next();
+  if (bag.jobStatusCode) return next();
+
+  var who = bag.who + '|' + _getOutputVersion.name;
+  logger.verbose(who, 'Inside');
+
+  bag.consoleAdapter.openGrp('Saving resource version');
+  bag.consoleAdapter.openCmd('Reading output version');
+
+  fs.readJson(bag.outputVersionFilePath,
+    function (err, outputVersion) {
+      // don't throw an error if this file doesn't exist
+      var msg;
+      if (err)
+        msg = util.format('Failed to read %s', bag.outputVersionFilePath);
+      else
+        msg = 'Successfully read output version';
+
+      bag.outputVersion = outputVersion || {};
+
+      bag.consoleAdapter.publishMsg(msg);
+      bag.consoleAdapter.closeCmd(true);
+
+      return next();
+    }
+  );
+}
+
+function _extendOutputVersionWithEnvs(bag, next) {
+  if (bag.isJobCancelled) return next();
+  if (bag.jobStatusCode) return next();
+
+  var who = bag.who + '|' + _extendOutputVersionWithEnvs.name;
+  logger.verbose(who, 'Inside');
+
+  bag.consoleAdapter.openCmd('Reading additional job properties');
+  var envFilePath = bag.stateDir + '/' + bag.inPayload.name + '.env';
+  var newVersionName = '';
+  var propertyBag = {};
+  try {
+    var envFile = fs.readFileSync(envFilePath).toString();
+    var lines = envFile.split('\n');
+    bag.consoleAdapter.publishMsg(
+      util.format('found file %s.  Checking for additional properties.',
+      envFilePath)
+    );
+    _.each(lines,
+      function (line) {
+        var nameAndValue = line.split('=');
+        var key = nameAndValue[0];
+        var value = nameAndValue[1];
+        if (key) {
+          bag.consoleAdapter.publishMsg('found a key: ' + key);
+          if (key === 'versionName')
+            newVersionName = value;
+          else
+            propertyBag[key] = value;
+        }
+      }
+    );
+  } catch (err) {
+    bag.consoleAdapter.publishMsg(
+      util.format('Could not parse file %s. Hence Skipping.',
+        envFilePath));
+    bag.consoleAdapter.publishMsg(
+      util.format('unable to read file %s.env', bag.inPayload.name));
+    bag.consoleAdapter.closeCmd(true);
+    bag.consoleAdapter.closeGrp(true);
+    return next();
+  }
+  var extraVersionInfo = {};
+  if (newVersionName) {
+    bag.consoleAdapter.publishMsg(
+      util.format('Found versionName %s'
```

Comment:
We can remove this. This is only used for runSh, I think a check was missed earlier.